src/execution/sema/sema_builtin.cpp

```diff
@@ -2420,17 +2477,59 @@ void Sema::CheckBuiltinParamCall(ast::CallExpr *call, ast::Builtin builtin) {
 
 void Sema::CheckBuiltinStringCall(ast::CallExpr *call, ast::Builtin builtin) {
   ast::BuiltinType::Kind sql_type;
+
+  // Checking to see if the first argument is an execution context
+  // All string functions have the execution context as their first argument
+  auto exec_ctx_kind = ast::BuiltinType::ExecutionContext;
+  if (!IsPointerToSpecificBuiltin(call->Arguments()[0]->GetType(), exec_ctx_kind)) {
+    ReportIncorrectCallArg(call, 0, GetBuiltinType(exec_ctx_kind)->PointerTo());
+    return;
+  }
+
   switch (builtin) {
-    case ast::Builtin::Chr: {
-      // check to make sure this function has two arguments
-      if (!CheckArgCount(call, 2)) {
+    case ast::Builtin::SplitPart: {
+      // check to make sure this function has four arguments
+      if (!CheckArgCount(call, 4)) {
         return;
       }
 
-      // checking to see if the first argument is an execution context
-      auto exec_ctx_kind = ast::BuiltinType::ExecutionContext;
-      if (!IsPointerToSpecificBuiltin(call->Arguments()[0]->GetType(), exec_ctx_kind)) {
-        ReportIncorrectCallArg(call, 0, GetBuiltinType(exec_ctx_kind)->PointerTo());
+      // checking to see if the second argument is a string
+      auto *resolved_type = Resolve(call->Arguments()[1]);
```

Comment:
Remove these calls to resolve. Remove the `resolved_type` variable altogether and just reference the arguments in `call->Arguments()`

Reply:
Done... I keep forgetting to check for resolve on these PRs